python/src/opendp/_extrinsics/microdata/universal_microdata_scheme.py

```diff
@@ -0,0 +1,166 @@
+
+        acceptance_criteria_meas: list[Measurement] = []
+        acceptance_criteria_thresholds: list[float] = []
+
+        # Quality Acceptance Criteria
+        if quality_epsilons is not None and quality_acceptance_criteria is not None:
+            (quality_acceptance_criteria_makers,
+             quality_accaptance_criteria_thresholds) = zip(*quality_acceptance_criteria)
+            assert (len(quality_acceptance_criteria_makers)
+                    == len(quality_accaptance_criteria_thresholds))
+
+            acceptance_criteria_meas = []
+            for make_private_ac, epsilon in zip(quality_acceptance_criteria_makers,
+                                                quality_epsilons):
+                acceptance_criteria_meas.append(dp.binary_search_chain(
+                        (lambda s: make_private_ac(input_domain, input_metric, output_measure,
+                                                   scale=s, reference_dataset=synth_dataset)),
+                        d_in=d_in, d_out=epsilon)
+                  )
+
+            acceptance_criteria_thresholds.extend(quality_accaptance_criteria_thresholds)
+
+        # Faithfulness Acceptance Criteria
+        if faithfulness_epsilon is not None and faithfulness_threshold is not None:
+            def faithfulness_meas(scale):
+                return make_private_faithfulness(input_domain,
+                                                 input_metric,
+                                                 output_measure,
+                                                 reference_dataset=synth_dataset,
+                                                 similarity=faithfulness_simiarlity,
+                                                 scale=scale)
+
+            def to_unfaithfulness(matching_cardinality):
+                return num_records - matching_cardinality
+
+            unfaithfulness_meas = dp.binary_search_chain(
+                lambda s: faithfulness_meas(s) >> to_unfaithfulness,
+                d_in=d_in, d_out=faithfulness_epsilon)
+
+            unfaithfulness_cardinality_threshold = int(np.ceil(faithfulness_threshold * num_records))
+
+            acceptance_criteria_meas.append(unfaithfulness_meas)
+            acceptance_criteria_thresholds.append(unfaithfulness_cardinality_threshold)
+
+        # Gather Results and Check if all Acceptance Criteria Passed
+        acceptance_criteria_results = [acceptance_criteria_pipeline_comp(meas)
+                                       for meas in acceptance_criteria_meas]
+
+        assert len(acceptance_criteria_results) == len(acceptance_criteria_thresholds)
+        all_acceptance_criteria_passed = all(value <= threshold
+                                             for value, threshold in zip(acceptance_criteria_results, acceptance_criteria_thresholds))
+
+        return all_acceptance_criteria_passed, configuration, synthesizer, synth_dataset, acceptance_criteria_results
```

Comment:
Return a namedtuple or something else that doesn't depend on the positions being stable.